src/components/search/FacetedFilter/FacetedFilter.tsx

```diff
@@ -0,0 +1,75 @@
+import { graphql } from 'gatsby'
+import React, { useState } from 'react'
+import { useSearch } from 'src/sdk/search/useSearch'
+import type { FacetedFilter_FacetsFragment } from '@generated/FacetedFilter_facets.graphql'
+
+import * as styles from './FacetedFilter.module.css'
+
+interface Props {
+  facets: FacetedFilter_FacetsFragment[]
+}
+
+function FacetedFilter({ facets }: Props) {
+  const [index, setIndex] = useState(-1)
+  const { toggleFacet } = useSearch()
+
+  return (
+    <div className={styles.container}>
+      {facets
+        .filter((facet) => facet.type === 'TEXT')
+        .map(({ name, values }, it) => (
+          <div key={`${name}-${it}`}>
+            <button
+              className={styles.button}
+              onClick={() => setIndex(it)}
+              data-testid="facet-filter-header"
+            >
+              {name}
+            </button>
+            <ul style={{ display: index !== it ? 'none' : 'block' }}>
+              {values?.map((item) => {
+                const id = `${name}-${item?.name}`
+
+                return (
+                  <li key={id}>
+                    <input
+                      id={id}
+                      type="checkbox"
+                      checked={!!item?.selected}
+                      onChange={() => toggleFacet(item as any)}
```

Comment:
yep, because the Type Facet is not exported [here](https://github.com/vtex-sites/base.store/blob/e1f716cd3770f4ec70fae3de5aefef7975f322f1/src/sdk/search/Provider.tsx#L52)

Reply:
it is because it's not using our graphql layer yet, but it's using IO's and IO's typings are broken

Reply:
Awesome!! 😄